vim25/json/discriminator_test.go

```diff
@@ -59,6 +61,58 @@ type DS8 struct {
 	F1 DS3 `json:"f1"`
 }
 
+type DS9 struct {
+	F1 int64
+}
+
+func (d DS9) MarshalJSON() ([]byte, error) {
```

Comment:
Can you please also write a test where the unmarshaler interface is not implemented by address? I ask because of:

```go
var unmarshalerType = reflect.TypeOf((*Unmarshaler)(nil)).Elem()

func supportsUnmarshaler(t reflect.Type) bool {
	return t.Implements(unmarshalerType) || reflect.PtrTo(t).Implements(unmarshalerType)
}
```

I am thinking something like the following:

```go
type DS10 struct {
	ds9 *DS9
}

func (d DS10) UnmarshalJSON(b []byte) error {
	if d.ds9 == nil {
		return nil
	}
	return d.ds9.UnmarshalJSON(b)
}
```

Since the helper func you wrote, `supportsUnmarshaler`, can act on both, there should be coverage for it.

Reply:
Will do.